packages/analytics-js-integrations/src/integrations/GainsightPX/browser.js

```diff
@@ -0,0 +1,123 @@
+/* eslint-disable class-methods-use-this */
+/* eslint-disable lines-between-class-members */
+import {
+  NAME,
+  DISPLAY_NAME,
+} from '@rudderstack/analytics-js-common/constants/integrations/GainsightPX/constants';
+import Logger from '../../utils/logger';
+import { loadNativeSdk } from './nativeSdkLoader';
+
+const logger = new Logger(DISPLAY_NAME);
+
+class GainsightPX {
+  constructor(config, analytics, destinationInfo) {
+    if (analytics.logLevel) {
+      logger.setLogLevel(analytics.logLevel);
+    }
+    this.analytics = analytics;
+    this.productKey = !config.productKey ? '' : config.productKey;
+    this.dataCenter = !config.dataCenter ? 'US' : config.dataCenter;
+    this.extraConfig = !config.extraConfig ? "{}" : config.extraConfig;
+    this.name = NAME;
+    ({
+      shouldApplyDeviceModeTransformation: this.shouldApplyDeviceModeTransformation,
+      propagateEventsUntransformedOnError: this.propagateEventsUntransformedOnError,
+      destinationId: this.destinationId,
+    } = destinationInfo ?? {});
+  }
+
+  init() {
+    loadNativeSdk(this.productKey, this.dataCenter, this.extraConfig);
+    this.initializeMe();
+  }
+
+  initializeMe() {
+    const userId = this.analytics.getUserId();
+
+    // Only proceed with identify if user ID is defined
+    if (userId) {
+      const visitorObj = { id: userId, ...this.analytics.getUserTraits() };
+
+      const accountObj = {
+        id: this.analytics.getGroupId(),
+        ...this.analytics.getGroupTraits(),
+      };
+
+      window.aptrinsic('identify', visitorObj, accountObj);
+    }
+  }
+
+  isLoaded() {
+    return !!(window.aptrinsic && window.aptrinsic.init);
+  }
+
+  isReady() {
+    return this.isLoaded();
+  }
+
+  /* utility functions --- Ends here ---  */
+
+  /*
+   * GainsightPX MAPPED FUNCTIONS :: identify, track, group
+   */
+
+  identify(rudderElement) {
+    let visitorObj = {};
+    let accountObj = {};
+    const { userId, groupId, context, traits } = rudderElement.message;
+    const id = userId;
+    const userTraits = context?.traits || {};
+    visitorObj = {
+      id,
+      ...userTraits,
+    };
+
+    const groupTraits = traits || {};
```

Comment:
In identify call, there will always be context.traits and no message.traits. message.traits is specifically group trait property, which is only considered during group call. Otherwise it will always return undefined. cc @MoumitaM

Reply:
Fixed